File: src/fr/geobert/radis/service/OnAlarmReceiver.java
package fr.geobert.radis.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class OnAlarmReceiver extends BroadcastReceiver {

	private Object startServiceLock;

	@Override
	public void onReceive(Context context, Intent intent) {		
		synchronized (startServiceLock) {
			RadisService.acquireStaticLock(context);
			context.startService(new Intent(context, RadisService.class));			
		}
	}

}

File: app/src/main/kotlin/fr/geobert/radis/service/OnAlarmReceiver.kt
package fr.geobert.radis.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

public class OnAlarmReceiver : BroadcastReceiver() {

    private val startServiceLock = Object()

    override fun onReceive(context: Context, intent: Intent) {
        synchronized (startServiceLock) {
            RadisService.acquireStaticLock(context)
            context.startService(Intent(context, RadisService::class.java))
        }
    }

}
